frontend/models/product-list/sections/index.ts

```diff
@@ -33,27 +41,92 @@ export const ProductListSectionSchema = z.union([
    LifetimeWarrantySectionSchema,
    RelatedPostsSectionSchema,
    FeaturedProductListsSectionSchema,
+   BannersSectionSchema,
+   SplitWithImageSectionSchema,
+   QuoteGallerySectionSchema,
+   PressQuotesSectionSchema,
 ]);
 
 interface ProductListSectionsArgs {
    strapiProductList: ProductListFieldsFragment | null | undefined;
+   reusableSections: ReusableSection[];
 }
 
 export function productListSections({
    strapiProductList,
+   reusableSections,
 }: ProductListSectionsArgs) {
    const sections: ProductListSection[] = [
       heroSection(),
       productListChildrenSection(),
       filterableProductsSection(),
    ];
+   const sectionsByPosition = getSectionsBucketByPosition(reusableSections);
+
+   insertSections({
+      from: sectionsByPosition['top'] ?? [],
+      into: sections,
+      after: 'top',
+   });
+
+   insertSections({
+      from: sectionsByPosition['after products'] ?? [],
+      into: sections,
+      after: 'FilterableProducts',
+   });
+
    const strapiSections = filterNullableItems(
       strapiProductList?.sections.map(productListSection)
    );
-   sections.push(...strapiSections);
+   insertSections({
+      from: strapiSections,
+      into: sections,
+      after: 'bottom',
+   });
+
+   insertSections({
+      from: sectionsByPosition['bottom'] ?? [],
+      into: sections,
+      after: 'bottom',
+   });
+
    return sections;
 }
 
+interface InsertSectionsArgs {
+   into: ProductListSection[];
+   from: ProductListSection[];
+   after: 'top' | 'bottom' | ProductListSection['type'];
+}
+
+function insertSections({ into, from, after }: InsertSectionsArgs): void {
+   if (after === 'top') {
+      into.splice(0, 0, ...from);
+   } else if (after === 'bottom') {
+      into.push(...from);
+   } else {
+      const index = into.findIndex((section) => section.type === after);
+      if (index !== -1) {
+         into.splice(index + 1, 0, ...from);
+      }
+   }
+}
+
+function getSectionsBucketByPosition(
+   reusableSections: ReusableSection[]
+): Record<ReusableSection['positionInProductList'], ProductListSection[]> {
+   const sectionsByPosition = groupBy(
+      reusableSections,
+      'positionInProductList'
+   );
```

Comment:
If we make a change like this, we can drop the `as any` below and gain a little type safety.

```suggestion
   const sectionsByPosition = groupBy(
      reusableSections,
      'positionInProductList'
   ) as Record<ReusableSection['positionInProductList'], ReusableSection[]>;
```